packages/core/src/components/MonthPicker/MonthPicker.jsx

```diff
@@ -0,0 +1,220 @@
+            key={month}
+            onChange={e => this.handleChange(e)}
+            className="ds-c-month-picker__month"
+            disabled={disabledMonths[i]}
+            inversed={inversed}
+            aria-label={this.monthsLong[i]}
+          >
+            {month}
+          </Choice>
+        ))}
+      </ul>
+    );
+  }
+
+  renderButton(text, onClick) {
+    return (
+      <Button
+        variation="primary"
+        size="small"
+        className="ds-u-margin-right--1"
+        onClick={onClick}
+        inversed={this.props.inversed}
+      >
+        {text}
+      </Button>
+    );
+  }
+
+  render() {
+    const { selectAllText, clearAllText } = this.props;
+    return (
+      <div className="ds-c-month-picker ds-u-margin-y--3">
+        <div className="ds-u-margin-bottom--3">
+          {this.renderButton(selectAllText, () => this.handleSelectAll())}
+          {this.renderButton(clearAllText, () => this.handleClearAll())}
+        </div>
+        {this.renderMonths(this.selectedMonths())}
+      </div>
+    );
+  }
+}
+
+MonthPicker.propTypes = {
+  /**
+   * The `input` field's `name` attribute
+   */
+  name: PropTypes.string.isRequired,
+  /**
+   * Locale string used to generate month names
+   */
+  locale: PropTypes.string,
+  /**
+   * Applies the "inverse" UI theme
+   */
+  inversed: PropTypes.bool,
+  /**
+   * Array of 12 booleans representing 12 months, where a value of `true`
+   * means the corresponding month is disabled.
+   */
+  disabledMonths: PropTypes.arrayOf(PropTypes.bool),
+  /**
+   * Array of 12 booleans representing 12 months, where a value of `true`
+   * means the corresponding month is selected.
+   * This will render a read-only field. If the field should be mutable,
+   * use `defaultSelectedMonths`.
+   */
+  selectedMonths: PropTypes.arrayOf(PropTypes.bool),
+  /**
+   * Array of 12 booleans representing 12 months, where a value of `true`
+   * means the corresponding month is selected by default.
+   * Sets the initial checked state for the 12 month checkboxes. Use this for
+   * an uncontrolled component; otherwise, use the `selectedMonths` property.
+   */
+  defaultSelectedMonths: PropTypes.arrayOf(PropTypes.bool),
+  onChange: PropTypes.func,
+  onSelectAll: PropTypes.func,
+  onClearAll: PropTypes.func,
+  /**
+   * For internationalization purposes, the text for the "Select all"
+   * button must be passed in as a prop.
+   */
+  selectAllText: PropTypes.string.isRequired,
+  /**
+   * For internationalization purposes, the text for the "Clear all"
+   * button must be passed in as a prop.
+   */
+  clearAllText: PropTypes.string.isRequired
+};
+
+export default MonthPicker;
+
+/**
+ * Generates an array of month names according to the given or default locale
+ *
+ * @param  {string} [locale] locale for generating month names
+ * @param  {boolean} [short] whether to return short month names
+ * @return {string[]}        array of short month names
```

Comment:
tiny nit: It wouldn't return short names if `short` is `false` right?

Reply:
Correct, and?

Reply:
ah, it says it returns "array of *short* month names". The tiniest of nits 🙈